### PR TITLE
DTFS2-4698 - GEF - UKEF submission maker_checker check

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue: 

* maker_checker users could edit a facility and submit to UKEF

# Changes made: 

 * every edit of an application updates the application to include the editor ID
 * ensures that maker_checker cannot submit to ukef (regardless of if or has not submitted to checker)
 * Updated tests
 * Added to e2e test to test many scenarios

### CODE REVIEW

File: e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js [hunk 1]
```diff
@@ -85,3 +91,352 @@ context('Submit application to UKEF', () => {
...
+
+      // login as a maker_checker and ensure that cannot return or submit to ukef
+      cy.login(CREDENTIALS.MAKER_CHECKER);
+      cy.visit(relative(`/gef/application-details/${dealIds[2]}`));
+      applicationPreview.returnButton().should('not.exist');
+      returnToMaker.submitButton().should('not.exist');
+    });
+  });
+
+  describe('BANK1_MAKER1 makes application, MakerChecker adds security questions only, MakerChecker should not be able to submit to ukef', () => {
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js:380]: Can we be consistent with the wording for maker and maker-checker, lots of different variants

-BANK1_MAKER1
- MAKER
- MAKER_CHECKER
- MakerChecker
- maker/checker

File: e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js [hunk 2]
```diff
@@ -85,3 +91,352 @@ context('Submit application to UKEF', () => {
...
+      applicationSubmission.confirmationPanelTitle();
+
+      // login as a maker_checker and ensure that cannot return or submit to ukef
+      cy.login(CREDENTIALS.MAKER_CHECKER);
+      cy.visit(relative(`/gef/application-details/${dealIds[2]}`));
+      applicationPreview.returnButton().should('not.exist');
+      returnToMaker.submitButton().should('not.exist');
+    });
+  });
+});
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js:442]: Can we split these tests up bit into seperate files?

File: gef-ui/server/controllers/confirm-cover-start-date/index.js [hunk 3]
```diff
@@ -15,8 +15,13 @@ const setError = (field, message) => validationErrorHandler({
   errMsg: message,
 });
 
-const updateCoverStartDate = async (facilityId, { coverStartDate, shouldCoverStartOnSubmission }) => {
+const updateCoverStartDate = async (facilityId, { coverStartDate, shouldCoverStartOnSubmission }, dealId, _id) => {
   try {
+    const applicationUpdate = {
+      editorId: _id,
```
Comment (ttbarnes) [inline gef-ui/server/controllers/confirm-cover-start-date/index.js:21]: Could we rename the _id param to `editorId` - would save doing this and make the params clearer 😄

File: gef-ui/server/controllers/confirm-cover-start-date/index.js [hunk 4]
```diff
@@ -30,6 +35,8 @@ const updateCoverStartDate = async (facilityId, { coverStartDate, shouldCoverSta
 
 const processCoverStartDate = async (req, res) => {
   const { dealId, facilityId } = req.params;
+  const { user } = req.session;
+  const { _id } = user;
```
Comment (ttbarnes) [inline gef-ui/server/controllers/confirm-cover-start-date/index.js:39]: Similar to above, if we call _id `userId` it'll make it clearer in the functions below. Feels a bit strange having `dealId, _id` down below

**Review by ttbarnes — COMMENTED** [review]
Few comments - great work on the `updateApplication` spy tests 👍

File: e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js:380]: done

File: gef-ui/server/controllers/confirm-cover-start-date/index.js (hunk 4 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/confirm-cover-start-date/index.js:39]: done

File: e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/scenarios/maker-should-not-submit-its-own-deals.spec.js:442]: done

File: e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker_checker-should-not-submit-edited-deal-add-document.spec.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import relative from '../../relativeURL';
```
Comment (IgnatG) [inline e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker_checker-should-not-submit-edited-deal-add-document.spec.js:1]: can we change the parent folder name to 
`maker-checker-user-role`

File: e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker-should-not-submit-its-own-submitted-deals.spec.js [hunk 6]
```diff
@@ -1,14 +1,15 @@
-import relative from '../relativeURL';
-import applicationDetails from '../pages/application-details';
-import automaticCover from '../pages/automatic-cover';
-import applicationSubmission from '../pages/application-submission';
-import CREDENTIALS from '../../fixtures/credentials.json';
-import applicationPreview from '../pages/application-preview';
-import returnToMaker from '../pages/return-to-maker';
-import statusBanner from '../pages/application-status-banner';
+import relative from '../../relativeURL';
```
Comment (IgnatG) [inline e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker-should-not-submit-its-own-submitted-deals.spec.js:1]: can we update the file name as well for this file to match the others from this folder?

File: gef-ui/server/controllers/about-facility/index.test.js [hunk 7]
```diff
@@ -124,6 +132,26 @@ describe('controllers/about-facility', () => {
       });
     });
 
+    it('calls api.updateApplication with editorId if successfully updates facility', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
+      mockRequest.query.saveAndReturn = 'true';
+      mockRequest.body['cover-start-date-day'] = format(tomorrow, 'd');
```
Comment (IgnatG) [inline gef-ui/server/controllers/about-facility/index.test.js:138]: this is specific for BSS. GEF should have `coverStartDate`

File: gef-ui/server/controllers/confirm-cover-start-date/index.js (hunk 3 above)
Comment (IgnatG) [reply gef-ui/server/controllers/confirm-cover-start-date/index.js:21]: I think @ttbarnes  meant `{ _id: editorId }` unless I'm mistaken?

File: e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker_checker-should-not-submit-edited-deal-add-document.spec.js (hunk 5 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker_checker-should-not-submit-edited-deal-add-document.spec.js:1]: done

File: e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker-should-not-submit-its-own-submitted-deals.spec.js (hunk 6 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker-should-not-submit-its-own-submitted-deals.spec.js:1]: done

Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/scenarios/Deal editors cannot submit to UKEF/maker-should-not-submit-its-own-submitted-deals.spec.js:1]: done

**Review by IgnatG — APPROVED** [review]

**Pull request merged**